From 389684083ab3740467023159ce85f55ece2782f9 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Thu, 26 Mar 2020 14:13:43 +0800
Subject: [PATCH 30/37] MLK-23681 arm64: dts: imx: enable pcieb on 8qm mek
 baseboard

commit 93def7a9870c429c12172639c015e04221bf5688 from
https://source.codeaurora.org/external/imx/linux-imx.git

Based on imx_4.1x kernel, enable the PCIEB on i.MX8QM MEK baseboard.
Regarding to the base board HW limitation(two Disable#) are not
connected. Only the standard PCIe EP device is supported on PCIEB port.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
(cherry picked from commit 6d92b251c8916e603325471ece2a6caa39d52fdc)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek.dtsi      | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
index c59e07f7cc36..5f4366f7431d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
@@ -460,6 +460,14 @@
 			>;
 		};
 
+		pinctrl_pcieb: pciebgrp {
+			fsl,pins = <
+				SC_P_PCIE_CTRL1_CLKREQ_B_LSIO_GPIO4_IO30	0x06000021
+				SC_P_PCIE_CTRL1_WAKE_B_LSIO_GPIO4_IO31		0x04000021
+				SC_P_PCIE_CTRL1_PERST_B_LSIO_GPIO5_IO00		0x06000021
+			>;
+		};
+
 		pinctrl_sim0: sim0grp {
 			fsl,pins = <
 				SC_P_SIM0_CLK_DMA_SIM0_CLK		0xc0000021
@@ -1136,6 +1144,15 @@
 	status = "okay";
 };
 
+&pcieb {
+	ext_osc = <1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pcieb>;
+	reset-gpio = <&gpio5 0 GPIO_ACTIVE_LOW>;
+	clkreq-gpio = <&gpio4 30 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
+
 &intmux_cm40 {
 	status = "okay";
 };
-- 
2.17.1

